addOns/authhelper/src/test/java/org/zaproxy/addon/authhelper/VerificationDetectionScanRuleUnitTest.java

```diff
@@ -19,12 +19,156 @@
  */
 package org.zaproxy.addon.authhelper;
 
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.Sets;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.MockedStatic;
+import org.parosproxy.paros.model.Session;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Context;
+
 /** Unit test for {@link VerificationDetectionScanRule}. */
 class VerificationDetectionScanRuleUnitTest
         extends PassiveScannerTest<VerificationDetectionScanRule> {
 
+    private static final Set<String> BASE_URLS =
+            Set.of("https://www.example.com/%s", "ttps://www.example.com/?action=%s");
+
     @Override
     protected VerificationDetectionScanRule createScanner() {
         return new VerificationDetectionScanRule();
     }
+
+    /* Provides URLs with poor candidate strings both as path components and query values */
+    private static Stream<String> providePoorCandidates() {
+        return BASE_URLS.stream()
+                .flatMap(url -> AuthUtils.POOR_CANDIDATE_INDICATORS.stream()
+                        // Format the URL with the indicator
+                        .map(url::formatted));
```

Comment:
I felt like I wouldn't remember exactly what this was doing 6months from now, hence the comment.